### PR TITLE
A test file for conductresist.py

### PR DESCRIPTION
**Pull request by FarBo**
This PR adds a test file for conductresist.py, which is including essential test scenarios for this module.

### CODE REVIEW

File: mqml/tests/test_conductresist.py [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+from mqml.parameter.conductresist import ConductResist
+from qcodes.instrument.base import Instrument
+import numpy as np
+
+@pytest.fixture(autouse=True)
+def close_all_instruments():
+    """Makes sure that after startup and teardown all instruments are closed"""
+    Instrument.close_all()
+    yield
+    Instrument.close_all()
```
Comment (jenshnielsen) [inline mqml/tests/test_conductresist.py:12]: Is this fixture needed. As far as I can see there are no instruments being created in these tests

Comment (FarBo) [reply mqml/tests/test_conductresist.py:12]: ConductResist is an Instrument.

Comment (jenshnielsen) [reply mqml/tests/test_conductresist.py:12]: Sorry I take that back. I can see that ConductResist is an instrument. It is unfortunate that it is found in a folder called parameters when it is not a Parameter.

File: mqml/tests/test_conductresist.py [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+
+def test_get_static_parameters():
+    "This tests the initial values of static parameters in the class"
```
Comment (jenshnielsen) [inline mqml/tests/test_conductresist.py:15]: What do you understand by static in this context?

File: mqml/tests/test_conductresist.py (hunk 1 above)
Comment (FarBo) [reply mqml/tests/test_conductresist.py:12]: No problem. The naming is not mature yet, you're right. I may perform some changes.

File: mqml/tests/test_conductresist.py (hunk 2 above)
Comment (FarBo) [reply mqml/tests/test_conductresist.py:15]: Should be changed to initial values.

File: mqml/tests/test_conductresist.py [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+    assert test_param.ACdiv() == 1e-4
+    assert test_param.DCdiv() == 1e-2
+
+def test_errors():
+    "This tests the erros"
+    lockin_param1 = 1
+    lockin_param2 = 2
+    test_param = ConductResist('test_param', lockin_param1=lockin_param1, lockin_param2=lockin_param2)
+
+    with pytest.raises(TypeError) as e:
```
Comment (jenshnielsen) [inline mqml/tests/test_conductresist.py:31]: You can use the match keyword with the pytest.raises context manager to simplify this

File: mqml/tests/test_conductresist.py [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+                            "test_param_resistance_fpm\')"
+
+def test_warnings():
+    "This tests warnings if zero divisions occur"
+    lockin_param1 = 1.0
+    lockin_param2 = 0.0
+    test_param = ConductResist('test_param', lockin_param1=lockin_param1, lockin_param2=lockin_param2)
+    test_param.GIamp(1e7)
+    test_param.GVamp(100.0)
+    nan = np.nan
```
Comment (jenshnielsen) [inline mqml/tests/test_conductresist.py:53]: this line seems to add minimum value over just comparing to np.nan

File: mqml/tests/test_conductresist.py [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+
```
Comment (jenshnielsen) [inline mqml/tests/test_conductresist.py:54]: Would be nice to use the pytest warning context manager here to assert that this raises a warning. https://docs.pytest.org/en/3.0.5/recwarn.html

**Review by jenshnielsen — APPROVED** [review]

**Comment by FarBo** [comment]
Thanks @jenshnielsen for the review.

File: mqml/tests/test_conductresist.py [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+def close_all_instruments():
+    """Makes sure that after startup and teardown, all instruments are closed"""
+    Instrument.close_all()
+    yield
+    Instrument.close_all()
+
+def test_get_initial_values():
+    "This tests the initial values of uncalculated parameters in the class"
+    lockin_param1 = 1
+    lockin_param2 = 2
```
Comment (jenshnielsen) [inline mqml/tests/test_conductresist.py:18]: This seems to still confuse parameters and the values of the parameters?

Comment (FarBo) [reply mqml/tests/test_conductresist.py:18]: That's right. I will have an another PR for this as I need to change the source file.

**Pull request merged**